samples/boards/bbc_microbit/pong/src/main.c

```diff
@@ -489,26 +492,32 @@ void pong_remote_lost(void)
 static void configure_buttons(void)
 {
 	static struct gpio_callback button_cb_data;
-	const struct device *gpio;
 
-	gpio = device_get_binding(DT_GPIO_LABEL(DT_ALIAS(sw0), gpios));
+	if (!device_is_ready(sw0_gpio.port)) {
+		printk("%s: device not ready.\n", sw0_gpio.port->name);
+		return;
+	}
+
+	if (!device_is_ready(sw1_gpio.port)) {
```

Comment:
It is not necessary to check if the device pointed by `sw1_gpio.port` is ready when it is required that the pointers `sw0_gpio.port` and `sw1_gpio.port` are equal.
Same story in the sound sample.

Reply:
There commits where not meant for this PR, will address this over in https://github.com/zephyrproject-rtos/zephyr/pull/47430